R/conversion.R

```diff
@@ -387,6 +387,35 @@ dim.pandas.core.frame.DataFrame <- function(x) {
     py_object_shape(x)
 }
 
+# Scipy sparse matrices
+
+#' @export
+dim.scipy.sparse.base.spmatrix <- function(x) {
+  if (py_is_null_xptr(x) || !py_available())
+    NULL
+  else
+    py_object_shape(x)
+}
+
+#' @export
+length.scipy.sparse.base.spmatrix <- function(x) {
+  if (py_is_null_xptr(x) || !py_available())
+    2L
```

Comment:
Could you test this as well to make sure that the length is correct when it's null pointer, e.g. use `saveRDS()` and `readRDS()`?

Reply:
How should `py_to_r` and `dim` behave when handed a null pointer? `dim` currently returns NULL and `py_to_r` throws an error, but not a well-behaved one.

Reply:
`NULL` seems reasonable. Here's similar handling for `pandas.Series`: https://github.com/rstudio/reticulate/blob/master/R/conversion.R#L219-L230
and for `pandas.DataFrame`: https://github.com/rstudio/reticulate/blob/master/R/conversion.R#L374-L388

Reply:
Okay, I've modified `length` to behave the same as for a dataframe. `py_to_r` gives uninformative errors for both scipy and pandas when handed a null pointer - not sure if this is outside the scope of this pull request.